docs/bulk-retries-design.md

```diff
@@ -18,22 +18,33 @@ When ServiceControl restarts, the in-memory representations are rebuilt by aggre
 
 When a retry batch is first created it has this state. This means that ServiceControl is finding and marking failed messages as belonging to this batch.
 
-To do this, a new document is created for each failed message. Each one has an id `FailedMessageRetry/{messageId}` and contains the failed message id and the retry batch id. As only one document with this key can exist at a time, this ensures that a Failed Message can only ever belong to a single batch. This document will exist until such a time as a new Failed Message with same Id comes through the error queue. This guarantees that there can only be one outstanding retry for a failed message at a time.
+To do this, a new document is created for each failed message. Each one has an id `FailedMessageRetry/{messageId}` and contains the failed message id and the retry batch id. As only one document with this key can exist at a time, this ensures that a Failed Message can only ever belong to a single batch. This FailedMessageRetry document will exist until such a time as a new Failed Message with same Id comes through the error queue. This guarantees that there can only be one outstanding retry for a failed message at a time.
 
 When all messages have been marked, a list of the `FailedMessageRetry` ids is appended to the batch and the batch changes status to `Staging`. The list inside of the Batch may contain failed messages which do not belong to this batch (because another batch claimed them in parallel). These will get filtered out during staging (below).
 
-When ServiceControl starts up it will attempt to adopt any batches that it finds in this status and move them to `Staging`. This can only happen if the SC process stops during the above process. Any documents that were already marked will be added to the batch. Any documents that had not yet been marked are ignored and will have to be retried again by the user. When SC starts up, it will generate a `Session ID` GUID. This GUID is stamped onto each new batch as it is created. This is how SC can tell if a batch is from a previous session and adopt it. Only Batches with a non-current session Id will be adotped by the orphan batch process.
+When ServiceControl starts up it will attempt to adopt (what does adopt mean?) any batches that it finds in this status and move them to `Staging`. This can only happen if the ServiceControl process stops during the above process. Any documents that were already marked will be added to the batch. Any documents that had not yet been marked are ignored and will have to be retried again by the user. (<- what does this mean?) When ServiceControl starts up, it will generate a `Session ID` GUID. This GUID is stamped onto each new batch as it is created. This is how ServiceControl can tell if a batch is from a previous session and adopt it. Only Batches with a non-current session Id will be adopted by the orphan batch process.
```

Comment:
> what does adopt mean?

We may be missing a concept here. When ServiceControl starts it will look for documents marked with a retry batch id, without a matching retry batch document. i.e. We started making the batch but we never completed it. We call retry batches in this state "orphaned" because the process that was assembling them has been lost.

When we "adopt" and orphaned batch, we take over the processing of it. We find all of the messages with the same missing retry batch id and create a retry batch with that id for them. This new retry batch document will then get picked up and processed like any other.

> Any documents that had not yet been marked are ignored and will have to be retried again by the user. (<- what does this mean?)

When you issue a bulk retry we start a background process that looks  for all of the messages that match your criteria and we assign them a retry batch id (in lots of 1000). If you kill the process before it is done, the details of what you were trying to retry are gone. Any documents we had already marked for retry will be retried. Any that were not marked for retry will require manual intervention to retry